### DISCUSSION TITLE
메소드와 멤버변수를 무슨 기준으로 은닉화하시나요?

### DISCUSSION THREAD

**Discussion by wonsss** [woowacourse-study/2022-code-review-study#16] [Lotto] [answered]
121 PR의 피드백 중에 무엇을 private 메소드로 해야 할지에 대한 내용이 있네요. 
[#121 PR](https://github.com/woowacourse/javascript-lotto/pull/121#discussion_r820185669)

저는 아직 명확히 감이 잘 오지 않는데, 여러분들의 생각이 궁금합니다. 😄

**Comment by lokba** [comment, marked as answer]
제가 다니는 학교에서, 교수가 이런말을 한 적이 있어요.

`원래 코드를 짤때, 전부 private로 시작해라` 저는 이 말이 되게 공감되었어요.

다 짜고 나서, private로 할지 public할지 고민을 하는게 아닌거죠.
우선, private로 구성을 다하고 외부에서 써야하는 친구일 경우 public을 하는거죠.

어떻게 생각하시나요? 🤔

**Reply by wonsss** [reply]
좋은 방법이네요. 록바!
그렇다면 "외부에서 접근해도 되는 메소드"는 어떤 기준을 갖고 판단할 수 있을까요?

**Reply by byhhh2** [reply]
오 그렇네요 록바

> public 으로 선언한 영역은 시간이 지나면서
내가 작성한 코드를 다른 사람들이 사용하는 영역.
즉, 외부와의 인터페이스 영역에 해당하는 것인데
이런 부분은 나중에 쉽게 바꿀 수가 없게 됩니다.
반면에 '내부 구현'은 언제나 바꿀 수 있도록 private이 보장해주는 것이죠.

검색하다가 요 글을 읽었는데요.
전 항상 `뭘 private로 하지?` 생각했던 것 같은데 사실상 `뭘 public으로 해줘야하지?` 이게 맞을지도 모르겠네요
